test/broadcast.jl

```diff
@@ -288,6 +288,8 @@ let identity = error, x = [1,2,3]
     x .= 1 # make sure it goes to broadcast!(Base.identity, ...), not identity
     @test x == [1,1,1]
 end
+# See issue #18176:
+@test ((a,b,c) -> a+b+c).(1.0:2, 3, 4) == ((a,b,c) -> a+b+c).(1.0:2, 3.0, 4.0) == [8,9]
```

Comment:
Given that this is also tested in #18200, instead of this, a test of the resulting expression after parsing would be more useful?